avro/src/test/java/com/fasterxml/jackson/dataformat/avro/interop/annotations/CustomSerializationTest.java

```diff
@@ -105,4 +106,61 @@ public void testUnionCustomDeSerialization() throws IOException {
 
         assertThat(result).isEqualTo(house);
     }
+
+    @JsonSerialize(using = NullValue.Serializer.class)
+    @JsonDeserialize(using = NullValue.Deserializer.class)
+    public final static class NullValue {
+        public NullValue() {
+        }
+
+        @Override
+        public boolean equals(final Object other) {
+            return other instanceof NullValue;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(getClass());
+        }
+
+        public static class Serializer extends JsonSerializer<NullValue> {
+            @Override
+            public void serialize(
+                final NullValue house,
+                final JsonGenerator generator,
+                final SerializerProvider serializers
+            ) throws IOException {
+                generator.writeNull();
+            }
+        }
+
+        public static class Deserializer extends JsonDeserializer<NullValue> {
+            @Override
+            public NullValue deserialize(
+                final JsonParser parser,
+                final DeserializationContext context
+            ) throws IOException {
+                parser.nextToken();
```

Comment:
I think this is not right: should instead check that current token is `JsonToken.VALUE_NULL`: parser already points to the first token of value to deserialize.